### ISSUE TITLE
bug: Postman Authorization error with a grant_type of client_credentials - no Access Token returned - Unsupported grant_type attribute: None

### ISSUE THREAD

**Issue by johnmchale** [localstack/localstack#6100]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Current Behavior

n.b. using the latest Docker image => localstack/localstack:latest

trying to use Localstack Pro  'AWS Cognito' service. Whilst using Postman and sending an authorization request to:
http://loclahost:4566/oauth2/token 
an error is returned stating:
Unsupported grant_type
attribute: None
(see image below)

n.b. the Client ID and Client Secret should be as shown in docker container log (n.b. created via setup.sh script run at docker startup) 

tested the same functionality with previous tagged images (i.e. 0.14.2 or 0.13.3) and they both work

<img width="1014" alt="image" src="https://user-images.githubusercontent.com/38525955/169390344-523bd032-c45e-4837-a5a5-88d82db75408.png">

would have expected to get the Access Token as shown below 

<img width="1004" alt="image" src="https://user-images.githubusercontent.com/38525955/169396483-7fd2be02-06d5-4083-b5d5-4f2d402a3035.png">

### Expected Behavior

would expect to get the Access Token as shown below 

<img width="1004" alt="image" src="https://user-images.githubusercontent.com/38525955/169396483-7fd2be02-06d5-4083-b5d5-4f2d402a3035.png">

### How are you starting LocalStack?

With a docker-compose file

### Steps To Reproduce

Obviously, you would need to enter a value for LOCALSTACK_API_KEY since AWS Cognito is not available in the FREE version of Localstack; 

```
version: '3.4'
services:
  localstack:
    container_name: LocalStack
    image: localstack/localstack:latest
    ports:
      - '4566-4597:4566-4597'
    environment:
      - LOCALSTACK_API_KEY=**********
      - SERVICES=cognito,sqs,s3
      - DEBUG=0
      - AWS_ACCESS_KEY=SOME_KEY
      - AWS_SECRET_KEY=SOME_KEY
      - AWS_DEFAULT_REGION=us-east-1
      - DOCKER_HOST=unix:///var/run/docker.sock
      - HOSTNAME_EXTERNAL=localhost
    volumes:
      - ./LocalStack.Setup:/docker-entrypoint-initaws.d/
      - /tmp/localstack:/tmp/localstack
      - /var/run/docker.sock:/var/run/docker.sock
```

also had a setup.sh script running under a ./Localstack.Setup folder to create user pool and client:
this is the content:

```
USER_POOL_ID=$(awslocal cognito-idp create-user-pool --region us-east-1 --pool-name test --query UserPool.Id --output text)
CLIENT_ID=$(awslocal cognito-idp create-user-pool-client --region us-east-1 --user-pool-id $USER_POOL_ID --client-name test-client --generate-secret --query UserPoolClient.ClientId --output text)

echo $(awslocal cognito-idp describe-user-pool-client --user-pool-id $USER_POOL_ID --client-id $CLIENT_ID --output json)
```

### Environment

```markdown
- OS: Docker Windows 10 
- LocalStack: latest
```

**Comment by whummer** [comment]
Hi @johnmchale , thanks for reporting this issue. We've pushed a couple of enhancements for the Cognito TOKEN endpoint now - can you please pull the `latest` Docker image and give it another try? Please keep us posted on how it goes.. Thanks!

**Comment by whummer** [comment]
Hi @johnmchale , quick follow-up - did you have a chance to give it another try with the `latest` Docker image? We've been adding a bunch of enhancements for Cognito, and also released our major version 1.0 of LocalStack in the meantime. Please let us know if the problem persists - happy to look into it in more detail.. Thanks!

**Comment by johnmchale** [comment]
I did; it didn't work
